art/attacks/evasion/pixel_threshold.py

```diff
@@ -0,0 +1,1346 @@
+                if count == limit - 1:
+                    break
+                else:
+                    continue
+            min_bounds = [0, 0]
+            for _ in range(img.shape[-1]):
+                min_bounds += [0]
+            min_bounds = min_bounds * limit
+            max_bounds = [img.shape[-3], img.shape[-2]]
+            for _ in range(img.shape[-1]):
+                max_bounds += [255]
+            max_bounds = max_bounds * limit
+            bounds = [min_bounds, max_bounds]
+        else:
+            bounds = [(0, img.shape[-3]), (0, img.shape[-2])]
+            for _ in range(img.shape[-1]):
+                bounds += [(0, 255)]
+            bounds = bounds * limit
+        return bounds, initial
+
+
+class ThresholdAttack(PixelThreshold):
+    """
+    This attack was originally implemented by Vargas et al. (2019).
+
+    | Paper link:
+        https://arxiv.org/abs/1906.06026
+    """
+
+    def __init__(
+            self,
+            classifier,
+            th=None,
+            es=0,
+            targeted=False,
+            verbose=False):
+        """
+        Create a :class:`.PixelAttack` instance.
+        """
+        super(
+            ThresholdAttack,
+            self).__init__(
+                classifier,
+                th,
+                es,
+                targeted,
+                verbose)
+        self.type_attack = 1
+
+    def perturb_image(self, x, img):
+        """
+        TODO: Write Comment
+        """
+        if x.ndim < 2:
+            x = np.array([x])
+        imgs = np.tile(img, [len(x)] + [1] * (x.ndim + 1))
+        x = x.astype(int)
+        for adv, image in zip(x, imgs):
+            for count, (i, j, k) in enumerate(
+                    product(range(image.shape[-3]), range(image.shape[-2]),
+                            range(image.shape[-1]))):
+                image[i, j, k] = adv[count]
+        return imgs
+
+    def get_bounds(self, img, limit):
+        """
+        TODO: Write Comment
+        """
+
+        def bound_limit(value):
+            return (
+                np.clip(
+                    value - limit,
+                    0,
+                    255),
+                np.clip(
+                    value + limit,
+                    0,
+                    255))
+
+        minbounds, maxbounds, bounds, initial = [], [], [], []
+
+        for i, j, k in product(range(img.shape[-3]), range(img.shape[-2]),
+                               range(img.shape[-1])):
+            initial += [img[i, j, k]]
+            bound = bound_limit(img[i, j, k])
+            if self.es == 0:
+                minbounds += [bound[0]]
+                maxbounds += [bound[1]]
+            else:
+                bounds += [bound]
+        if self.es == 0:
+            bounds = [minbounds, maxbounds]
+
+        return bounds, initial
+
+
+# TODO: Make the attack compatible with current version of SciPy Optimize
+# Differential Evolution
```

Comment:
Is this comment required?

Reply:
Yes, The current version of Scipy's differential evolution doesn't work with the code and the following code which is a modification of Scipy's code works. However, it may be possible someone identifies the difference in the implementation and can resolve the usage by simply using the scipy's import.

Reply:
Actually the code uses a modified script of SciPy's implementation of Differential Evolution Optimisation. However, the best case would be to make the DE optimisation compatible with the current SciPy's implementation.
At the current stage, I have not yet identified all the changes in the implementation but it can be done at a later stage. 

_Please let it be here so that later I or someone else can update this._

Reply:
Can we combine explanation with the comment above to be add in line 45?

Reply:
Yeah Sure. That would be an ideal case I think.

Reply:
Ok.